Entities.js

```diff
@@ -0,0 +1,339 @@
+import { StyleSheet, Text, View, Image } from 'react-native';
+import { Dimensions, Keyboard } from 'react-native';
+import { React, Component, useEffect, useState, useRef } from 'react';
+import { UseOnKeyPress, UseOnKeyRelease } from "./Key.js"
+import { CreateCircle, CreateBrick, DeleteEntity } from "./Dodge.js"
+import Matter from "matter-js"
+
+const MAXwidth = Dimensions.get('window').width;
+const MAXheight = Dimensions.get('window').height;
+let counter = 0
+let GlobalX = 400
+let GlobalY = 400
+let playerHeight = 150
+let playerWidth = 150
+let input = [0,0,0,0]
+//      left,right,up,down (w,a,s,d) each stores a value, 0 = not pressed
+
+//count intersections with players
+const Counter = () => {
+  console.log({counter})
+} 
+
+//function to detect wasd input on computer
+function MoveKnight (){
```

Comment:
Maybe move this to Dodge.js? Have it set ents["knight"].input[0] = 1.5

It does affect an Entity, but its really more about what affect the keyboard input has. e.g. I can imagine re-using the Knight in a different game, where the controls are different (or run by an AI or whatever).

Reply:
Also, I think the same effect might be more intuitive if you thought of it as "acceleration". i.e.
When "left" is pressed: acceleration[0] -= 1.5; When "left" is released: acceleration[0] += 1.5
When "right" is pressed: acceleration[0] += 1.5; etc
(Your version is technically safer, since it will fix itself if the browser "forgets" a keyup)


[This is kind of overkill, but might be interesting]
Alternatively: have this function just record what interesting keys are currently down:
```
// Listens for interesting key presses, recording which ones are currently down in "input"
// NB: you could even simplify this and just keep a dictionary of *every* key which is currently down
function ListenForKeys(keys) {
  // One entry for each interesting key, they all start "up"
  for (key in keys) {
     input[key] = False
  }
  document.addEventListener('keydown', (event) => {
        if(event.key in input) {  // this is an interesting key
          input[event.key] = True;
        }
   });
   document.addEventListener('keyup', (event) => {
        if(event.key in input) {  // this is an interesting key
          input[event.key] = False;
        }
   });
```

Elsewhere, at the start of the program:
```
ListenForKeys(['w', 'a', 's', 'd']);
```

Elsewhere, maybe in the game loop:
```
if (input['w']) { knight.acceleration[1] -= 1.5; }
```

Reply:
I would but it was a huge hassle to get it to work at all, the error culprit:
yOu maY bE BreaKing THe rULEs of hOoKs
I really don't want to deal with that again unless it is necessary

Reply:
Yeah I don't think it's interesting or important to struggle with input, it's not surprising to me that it's a place with pain points.

I would still suggest a small change, where maybe instead of input[<number 1 to 4>] you use input[<key that was pressed, e.g. 'w' or 'left'>] just to make the meaning more clear, without changing the functionality.

Later in the move function it becomes
```
knight.velocity.x = knight.velocity.x + input['right'] - input['left']
```

Rather than
```
knight.velocity.x = knight.velocity.x + input[1] - input[0] // how do you know it's not input[2] - input[0] or input[0] - input[1]?
```